src/rollup/index.ts

```diff
@@ -344,7 +361,149 @@ export default function rollup (rawInputOptions: GenericConfigObject) {
+						name: string,
+						imports: string[],
+						exports: string[],
+						modules: ModuleJSON[]
+					}
+				} = {};
+				Object.keys(bundles).forEach(bundleName => {
+					const bundle = bundles[bundleName];
+
+					chunkOutputBundles[bundleName] = {
+						name: bundleName,
+						imports: bundle.getImportIds(),
+						exports: bundle.getExportNames(),
+						modules: bundle.getJsonModules()
+					};
+				});
+
+				function generate (rawOutputOptions: GenericConfigObject) {
+					const outputOptions = getAndCheckOutputOptions(inputOptions, rawOutputOptions);
+
+					if (outputOptions.format === 'umd' || outputOptions.format === 'iife') {
+						error({
+							code: 'INVALID_OPTION',
+							message: 'UMD and IIFE output formats are not supported with the experimentalCodeSplitting option.'
+						});
+					}
+
+					timeStart('--GENERATE--');
+
+					const generated: { [chunkName: string]: SourceDescription } = {};
+
+					const promise = Promise.all(Object.keys(bundles).map(bundleName => {
+						const bundle = bundles[bundleName];
+						return bundle.render(outputOptions)
+							.then(rendered => {
+								timeEnd('--GENERATE--');
+
+								graph.plugins.forEach(plugin => {
+									if (plugin.ongenerate) {
+										const bundle = chunkOutputBundles[bundleName];
+										plugin.ongenerate(assign({ bundle }, outputOptions), rendered);
+									}
+								});
+
+								flushTime();
+
+								generated[bundleName] = rendered;
+							});
+					}))
+						.then(() => {
+							return generated;
+						});
+
+					Object.defineProperty(promise, 'code', throwAsyncGenerateError);
+					Object.defineProperty(promise, 'map', throwAsyncGenerateError);
+
+					return promise;
+				}
+
+				return {
+					chunks: chunkOutputBundles,
+					generate,
+					write (outputOptions: OutputOptions) {
+						if (!outputOptions || !outputOptions.dir) {
+							error({
+								code: 'MISSING_OPTION',
+								message: 'You must specify output.dir for multiple inputs'
+							});
+						}
+
+						return generate(outputOptions).then(result => {
+							const dir = outputOptions.dir;
+
+							return Promise.all(Object.keys(result).map(chunkName => {
+								let chunk = result[chunkName];
+								let { code, map } = chunk;
+
+								const promises = [];
+
+								if (outputOptions.sourcemap) {
+									let url;
+
+									if (outputOptions.sourcemap === 'inline') {
+										url = (<any>map).toUrl();
+									} else {
+										url = `${chunkName}.map`;
+										promises.push(writeFile(dir + '/' + chunkName + '.map', map.toString()));
+									}
+
+									code += `//# ${SOURCEMAPPING_URL}=${url}\n`;
+								}
+
+								promises.push(writeFile(dir + '/' + chunkName, code));
+								return Promise.all(promises).then(() => {
+									return mapSequence(
+										graph.plugins.filter(plugin => plugin.onwrite), (plugin: Plugin) =>
+											Promise.resolve(plugin.onwrite(assign({ bundle: chunk }, outputOptions), chunk))
```

Comment:
Should `onwrite` also receive the `chunkName` as a parameter somewhere? Previously in "single bundle mode", `onwrite` could figure out what files were written using the `outputOptions`, but now with chunks, I think it needs the `chunkName` to be able to re-construct the written file paths.

Reply:
The chunk object above actually does include the `name` as well, so I believe that would be passed to the plugins ok?

Reply:
Ah okay, must've misread it. Sounds good!